BlackCubes/covidmap-react-native
# Map Component and Map Layout

## Changes
1. Created a `MapComponent` that renders the map with `react-native-maps`.
2. Created a `MapLayout` that houses `MapComponent`, `SearchBar`, and `DrawerButton` to be used for functionalities.
3. Refactored `DrawerMenu` to render the `MapLayout`.

## Purpose
There should be a Map component that just displays the map, and a Map Layout that houses the main functionalities.

## Approach
The `MapComponent` simply displays the map from `react-native-maps` with any custom stylings from `styled-components`, and the props it takes in are `mapviewHeight`, `mapviewRegion`, and `mapviewWidth`. The `mapviewHeight` and `mapviewWidth` are needed in order to display the map (without these, the map would not display), and the `mapviewRegion` is the display location of the map (in this case, it is San Francisco).

The `MapLayout` houses anything related to map and its functionalities such as the `SearchBar` (since it only appears when the map is displayed), the `MapComponent`, and the `DrawerButton` (this would actually be stored in the `/vaccine-layout` since it would appear, and so in the next issue this component would be stored in `/commons`).

Last of all was the refactorization of the `DrawerMenu` to display `MapLayout` in `"World"`, `"Country Province Stats"`, `"US Total"`, `"State Counties Totals"`, `"World Vaccination Totals"`, and `"US Vaccination Total"` while removing some of the test components.

## Testing
1. Pull changes.
2. On your terminal, go to the `/covid-mapper` directory and run expo `expo start`.
3. Once expo starts, choose a simulator of your choice and open the app.
4. Make sure the map appears first.
5. Open the drawer menu and click on the links making sure the map always appears.
6. Click on the floating icon on the bottom right and make sure the drawer menu opens.

## Screenshots

https://user-images.githubusercontent.com/29642735/137820254-795d4d10-60c5-4a3d-ab07-3b7f8b5ff209.mp4



Closes #23
